src/tools/openapi.ts

```diff
@@ -93,10 +108,16 @@ Only use the OpenAPI tool if you need to communicate to external servers.`;
                       .filter((p: any) => p.required === true)
                       .map((p: any) => p.name),
                     properties: methodSpec.parameters.reduce(
-                      (acc: any, p: any) => ({
-                        ...acc,
-                        [p.name]: { ...p.schema, description: p.name },
-                      }),
+                      (acc: any, p: any) =>
+                        !p.$ref
+                          ? { ...acc, [p.name]: { ...p.schema, description: p.name } }
+                          : {
+                              ...acc,
+                              [getReferencedObject(this.openApiSchema, p.$ref)?.name]: {
+                                ...getReferencedObject(this.openApiSchema, p.$ref)?.schema,
+                                description: getReferencedObject(this.openApiSchema, p.$ref)?.name,
+                              },
+                            },
```

Comment:
Isn't there any open-source library that can do the conversion?

Reply:
There are the openapi validators. I looked for the library that resolves the reference but I couldn't fine one.  It's probably only for the openapi schema.